### PR TITLE
Use new params in XrayCentre

### PR DESCRIPTION
**Pull request by dperl-dls**
Building on #1330 uses the new params for the xraycentre plans

Towards #1289 

### To test:
1. Run tests

### CODE REVIEW

File: tests/test_data/new_parameter_json_files/test_gridscan_param_defaults.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    "y_start_um": 0.0,
+    "y2_start_um": 0.0,
+    "z_start_um": 0.0,
+    "z2_start_um": 0.0,
+    "detector_distance_mm": 100.0,
+    "omega_start_deg": 0.0,
+    "exposure_time_s": 0.1,
+    "set_stub_offsets": true,
+    "ispyb_extras": {
+        "microns_per_pixel_x": 1.25,
```
Comment (rtuck99) [inline tests/test_data/new_parameter_json_files/test_gridscan_param_defaults.json:32]: Ditto comments in previous PR re microns_per_pixel

File: src/hyperion/external_interaction/callbacks/grid_detection_callback.py [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+    y2_start_um: float
+    z_start_um: float
+    z2_start_um: float
+    x_steps: int
+    y_steps: int
+    z_steps: int
+    x_step_size_um: float
+    y_step_size_um: float
+    z_step_size_um: float
+    set_stub_offsets: bool
```
Comment (DominicOram) [inline src/hyperion/external_interaction/callbacks/grid_detection_callback.py:27]: Should: Some of these parameters don't change from the grid detection so I don't think we need to have them in here at all. I can't see any reasonable way that we would want to change them based on an optical detection in the near future either:
* `set_stub_offsets`
* `transmission_frac`
* `exposure_time_s`

File: src/hyperion/external_interaction/callbacks/grid_detection_callback.py [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+
+
 class GridDetectionCallback(CallbackBase):
```
Comment (DominicOram) [inline src/hyperion/external_interaction/callbacks/grid_detection_callback.py:30]: Should: I don't think `run_up_distance_mm` is ever used here now so we can remove

File: src/hyperion/parameters/gridscan.py [hunk 4]
```diff
@@ -94,6 +97,7 @@ def detector_params(self):
         ), "Detector distance must be filled before generating DetectorParams"
         os.makedirs(self.storage_directory, exist_ok=True)
         return DetectorParams(
+            detector_size_constants=self.detector,  # type: ignore
```
Comment (DominicOram) [inline src/hyperion/parameters/gridscan.py:100]: Could: Can you put a comment here that this should get tidied by https://github.com/DiamondLightSource/hyperion/issues/1307

**Review by DominicOram — CHANGES_REQUESTED** [review]
Awesome, this is so much cleaner, particularly the `UpdateParams`. Some comments to make it a step cleaner but otherwise good.

**Review by DominicOram — APPROVED** [review]
Great, thanks. I've added a commit to take it a step further and remove stub offsets and exposure time as they're not used. Otherwise all looks good now!

**Pull request merged**